apps/site/src/pages/api/blocks/create.api.ts

```diff
@@ -62,43 +62,81 @@ export default createAuthenticatedHandler<
 
     const { blockName: untransformedBlockName, npmPackageName } = req.body;
 
-    const slugifiedBlockName = generateSlug(untransformedBlockName);
+    const { rawBlockNamespace, rawBlockNameWithoutNamespace } =
+      untransformedBlockName.match(
+        /^(@(?<rawBlockNamespace>[a-z0-9]+(?:(?:-|_)+[a-z0-9]+)*)\/)?(?<rawBlockNameWithoutNamespace>[a-z0-9]+(?:(?:-|_)+[a-z0-9]+)*)$/,
+      )?.groups ?? {};
 
-    if (slugifiedBlockName !== untransformedBlockName) {
+    if (!rawBlockNameWithoutNamespace) {
       return res.status(400).json(
         formatErrors({
-          msg: `Block name '${untransformedBlockName}' must be a slug. Try ${slugifiedBlockName} instead`,
-          code: "NAME_TAKEN",
+          msg: `Block name must be a slug or defined as '@namespace/block-name' (all lowercase). Current value: '${untransformedBlockName}'`,
+          code: "INVALID_INPUT",
+        }),
+      );
+    }
+
+    const blockName = createPathWithNamespace(
+      rawBlockNameWithoutNamespace,
+      rawBlockNamespace || shortname,
+    );
+
+    const canonicalBlockNameWithoutNamespace = generateSlug(
+      rawBlockNameWithoutNamespace,
+    );
+
+    const canonicalBlockName = createPathWithNamespace(
+      canonicalBlockNameWithoutNamespace,
+      generateSlug(rawBlockNamespace || shortname),
+    );
+
+    if (canonicalBlockName !== blockName) {
+      if (!rawBlockNamespace) {
+        return res.status(400).json(
+          formatErrors({
+            msg: `Block name '${untransformedBlockName}' must be a slug. Try ${canonicalBlockNameWithoutNamespace} instead`,
+            code: "INVALID_INPUT",
+          }),
+        );
+      }
+      return res.status(400).json(
+        formatErrors({
+          msg: `Block name '${untransformedBlockName}' does not match its canonical representation. Try ${canonicalBlockName} instead`,
+          code: "INVALID_INPUT",
+        }),
+      );
+    }
+
+    if (rawBlockNamespace && rawBlockNamespace !== shortname) {
+      return res.status(400).json(
+        formatErrors({
+          msg: `Unable to publish '${untransformedBlockName}' because the token belongs to '${shortname}' and not '${rawBlockNamespace}'`,
```

Comment:
Although this is slightly less helpful to the genuine user, it avoids introducing a way of someone who finds this API key discovering who it belongs to (there is no way of doing so at the moment other than by trial and error – this would make it quicker to exploit). If you agree with this change, it also applies to other files.
```suggestion
          msg: `Unable to publish '${untransformedBlockName}' because the API key provided does not belong to '${rawBlockNamespace}'`,
```

Reply:
https://github.com/blockprotocol/blockprotocol/pull/953/commits/ef12fe11a05a0316f4825d92a64060a84540c52e + https://github.com/blockprotocol/blockprotocol/pull/953/commits/17a9685a251bf103a34ca79516a198c789ba7d01